### PR TITLE
fix: Move firebase init in app.component init

### LINKED ISSUE #121: Sample not loading (loading indicator never ends)
Sample not loading (loading indicator never ends).

It happens with next versions and it was detected on Mar 30, 2018 3:35 AM
It was last ok on Mar 29, 2018 2:17 AM

Happens only on `template-master-detail-ng`, no issues with JS and TS.

Package.json:
```
{
  "description": "NativeScript Application",
  "license": "SEE LICENSE IN <your-license-filename>",
  "readme": "NativeScript Application",
  "repository": "<fill-your-repository-here>",
  "nativescript": {
    "id": "org.nativescript.templatemasterdetailng",
    "tns-android": {
      "version": "4.0.0-2018.3.22.2"
    },
    "tns-ios": {
      "version": "4.0.1-2018-03-26-01"
    }
  },
  "scripts": {
    "lint": "tslint \"app/**/*.ts\"",
    "update-ng": "update-app-ng-deps"
  },
  "dependencies": {
    "@angular/animations": "~5.2.0",
    "@angular/common": "~5.2.0",
    "@angular/compiler": "~5.2.0",
    "@angular/core": "~5.2.0",
    "@angular/forms": "~5.2.0",
    "@angular/http": "~5.2.0",
    "@angular/platform-browser": "~5.2.0",
    "@angular/platform-browser-dynamic": "~5.2.0",
    "@angular/router": "~5.2.0",
    "nativescript-angular": "5.3.0-2018-04-02-02",
    "nativescript-imagepicker": "5.0.0",
    "nativescript-plugin-firebase": "5.0.5",
    "nativescript-theme-core": "1.0.4",
    "nativescript-ui-listview": "3.5.0",
    "reflect-metadata": "0.1.10",
    "rxjs": "^5.5.0",
    "tns-core-modules": "next",
    "zone.js": "^0.8.4"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~5.2.0",
    "@ngtools/webpack": "~1.9.4",
    "babel-traverse": "6.26.0",
    "babel-types": "6.26.0",
    "babylon": "6.18.0",
    "clean-webpack-plugin": "~0.1.19",
    "codelyzer": "4.0.2",
    "copy-webpack-plugin": "~4.3.0",
    "css-loader": "~0.28.7",
    "extract-text-webpack-plugin": "~3.0.2",
    "lazy": "1.0.11",
    "nativescript-dev-sass": "1.3.5",
    "nativescript-dev-typescript": "0.6.0",
    "nativescript-dev-webpack": "0.10.0-2018-04-02-01",
    "nativescript-worker-loader": "~0.8.1",
    "raw-loader": "~0.5.1",
    "resolve-url-loader": "~2.2.1",
    "sass-loader": "~6.0.6",
    "tslint": "5.8.0",
    "typescript": "~2.6.2",
    "uglifyjs-webpack-plugin": "~1.1.6",
    "webpack": "~3.10.0",
    "webpack-bundle-analyzer": "^2.9.1",
    "webpack-sources": "~1.1.0"
  }
}
```

Build steps:
```
tns create template-master-detail-ng --template https://github.com/NativeScript/template-master-detail-ng#master
tns build android --release --bundle --env.uglify --env.aot  --env.snapshot <all the signing options>
```

### PR DESCRIPTION
**Pull request by vakrilov**
Fixes #121

### CODE REVIEW

**Comment by manoldonev** [comment]
run ci

File: package.json [hunk 1]
```diff
@@ -36,9 +36,9 @@
     "@angular/platform-browser": "~5.2.0",
     "@angular/platform-browser-dynamic": "~5.2.0",
     "@angular/router": "~5.2.0",
-    "nativescript-angular": "~5.2.0",
+    "nativescript-angular": "next",
```
Comment (dtopuzov) [inline package.json]: Can we stick with versions like `~5.2.0`?

**Comment by manoldonev** [comment]
@dtopuzov maybe this should be discussed on retrospective meeting as well -- the reasoning now is that master is a development branch and we could not migrate the template (make it ready for {N} 4.0) unless we reference "next" dependencies during development.

**Comment by manoldonev** [comment]
run ci

**Pull request merged**